File: 7-ReverseInteger/7-ReverseInteger.js
// Last updated: 24/04/2026, 10:06:02
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let sign = x < 0 ? -1 : 1;
    x = Math.abs(x);

    let rev = 0;
    while (x > 0) {
        rev = rev * 10 + (x % 10);
        x = Math.floor(x / 10);
    }

        rev = rev * sign;

    if (rev < -2147483648 || rev > 2147483647) {
        return 0;
    }

    return rev;
};

console.log(reverse(-123));

File: 7-ReverseInteger/7-ReverseInteger.cpp
// Last updated: 06/06/2026, 11:51:48
1class Solution {
2public:
3    int reverse(int x) {
4       long long rev = 0;
5       while(x){
6        rev = rev * 10+ x%10;
7        x /= 10;
8       } 
9
10       return (rev>INT_MAX || rev<INT_MIN)? 0 : rev;
11    }
12};